x-pack/plugins/ml/public/application/explorer/explorer_utils.ts

```diff
@@ -26,34 +26,102 @@ import {
 import { parseInterval } from '../../../common/util/parse_interval';
 import { ml } from '../services/ml_api_service';
 import { mlJobService } from '../services/job_service';
-import { getTimeBucketsFromCache } from '../util/time_buckets';
-import { getTimefilter, getUiSettings } from '../util/dependency_cache';
+import { getUiSettings } from '../util/dependency_cache';
 
 import {
   MAX_CATEGORY_EXAMPLES,
   MAX_INFLUENCER_FIELD_VALUES,
   SWIMLANE_TYPE,
+  SwimlaneType,
   VIEW_BY_JOB_LABEL,
 } from './explorer_constants';
+import type { CombinedJob } from '../../../common/types/anomaly_detection_jobs';
+import { MlResultsService } from '../services/results_service';
+import { InfluencersFilterQuery } from '../../../common/types/es_client';
+import { TimeRangeBounds } from '../util/time_buckets';
+import { Annotations, AnnotationsTable } from '../../../common/types/annotations';
+import { Influencer } from '../../../common/types/anomalies';
+import { RecordForInfluencer } from '../services/results_service/results_service';
+
+export interface ExplorerJob {
+  id: string;
+  selected: boolean;
+  bucketSpanSeconds: number;
+}
+
+interface ClearedSelectedAnomaliesState {
+  selectedCells: undefined;
+}
+
+export interface SwimlanePoint {
```

Comment:
As these new interfaces are exported, should consider renaming to `SwimLanePoint` and `SwimLaneData`, but we have a lot of usage of `Swimlane` still, so maybe not worth the code churn?

Reply:
As discussed, this is best left for a follow up